File: org.summer.view.widget/src/org/summer/view/widget/data/BindingFlags.java
package org.summer.view.widget.data;
    // Flags indicating run-time properties of a BindingExpression 
//    [Flags]
    /*internal*/public enum BindingFlags //: uint
    {
        // names used by Binding 

        OneWay                  ,//= PrivateFlags.iSourceToTarget, 
        TwoWay                  ,//= PrivateFlags.iSourceToTarget | PrivateFlags.iTargetToSource, 
        OneWayToSource          ,//= PrivateFlags.iTargetToSource,
        OneTime                 ,//= 0, 
        PropDefault             ,//= PrivateFlags.iPropDefault,
        NotifyOnTargetUpdated   ,//= PrivateFlags.iNotifyOnTargetUpdated,
        NotifyOnSourceUpdated   ,//= PrivateFlags.iNotifyOnSourceUpdated,
        NotifyOnValidationError ,//= PrivateFlags.iNotifyOnValidationError, 
        UpdateOnPropertyChanged ,//= 0,
        UpdateOnLostFocus       ,//= PrivateFlags.iUpdateOnLostFocus, 
        UpdateExplicitly        ,//= PrivateFlags.iUpdateExplicitly, 
        UpdateDefault           ,//= PrivateFlags.iUpdateDefault,
        PathGeneratedInternally ,//= PrivateFlags.iPathGeneratedInternally, 
        ValidatesOnExceptions   ,//= PrivateFlags.iValidatesOnExceptions,
        ValidatesOnDataErrors   ,//= PrivateFlags.iValidatesOnDataErrors,
        ValidatesOnNotifyDataErrors ,//= PrivateFlags.iValidatesOnNotifyDataErrors,

        Default                 ,//= PropDefault | UpdateDefault,

        /// <summary> Error value, returned by FlagsFrom to indicate faulty input</summary> 
        IllegalInput                ,//= PrivateFlags.iIllegalInput,

        PropagationMask ,//= OneWay | TwoWay | OneWayToSource | OneTime | PropDefault,
        UpdateMask      ,//= UpdateOnPropertyChanged | UpdateOnLostFocus | UpdateExplicitly | UpdateDefault,
    }

File: summer/WebContent/js/system/windows/data/BindingFlags.js
/**
 * BindingFlags
 */
define(["dojo/_base/declare", "data/PrivateFlags", "system/Type"], 
		function(declare, PrivateFlags, Type){

    var BindingFlags = declare("BindingFlags", Object,{ 
    });	


	BindingFlags.OneWay                  = PrivateFlags.iSourceToTarget; 
	BindingFlags.TwoWay                  = PrivateFlags.iSourceToTarget | PrivateFlags.iTargetToSource; 
	BindingFlags.OneWayToSource          = PrivateFlags.iTargetToSource;
	BindingFlags.OneTime                 = 0; 
	BindingFlags.PropDefault             = PrivateFlags.iPropDefault;
	BindingFlags.NotifyOnTargetUpdated   = PrivateFlags.iNotifyOnTargetUpdated;
	BindingFlags.NotifyOnSourceUpdated   = PrivateFlags.iNotifyOnSourceUpdated;
	BindingFlags.NotifyOnValidationError = PrivateFlags.iNotifyOnValidationError;
	BindingFlags.UpdateOnPropertyChanged = 0;
	BindingFlags.UpdateOnLostFocus       = PrivateFlags.iUpdateOnLostFocus;
	BindingFlags.UpdateExplicitly        = PrivateFlags.iUpdateExplicitly;
	BindingFlags.UpdateDefault           = PrivateFlags.iUpdateDefault;
	BindingFlags.PathGeneratedInternally = PrivateFlags.iPathGeneratedInternally; 
	BindingFlags.ValidatesOnExceptions   = PrivateFlags.iValidatesOnExceptions;
	BindingFlags.ValidatesOnDataErrors   = PrivateFlags.iValidatesOnDataErrors;
	BindingFlags.ValidatesOnNotifyDataErrors = PrivateFlags.iValidatesOnNotifyDataErrors;
 
	BindingFlags.Default                 = BindingFlags.PropDefault | BindingFlags.UpdateDefault;
 
    /// <summary> Error value, returned by FlagsFrom to indicate faulty input</summary> 
	BindingFlags.IllegalInput                = PrivateFlags.iIllegalInput;
 
	BindingFlags.PropagationMask = BindingFlags.OneWay | BindingFlags.TwoWay | 
									BindingFlags.OneWayToSource | BindingFlags.OneTime | BindingFlags.PropDefaul;
	BindingFlags.UpdateMask      = BindingFlags.UpdateOnPropertyChanged | BindingFlags.UpdateOnLostFocus | 
									BindingFlags.UpdateExplicitly | BindingFlags.UpdateDefault;


	BindingFlags.Type = new Type("BindingFlags", BindingFlags, [Object.Type]); 
    return BindingFlags;
        
});